synpurge/cli.py

```diff
@@ -226,3 +229,51 @@ def find_event_id(room_id, upto, token=None):
                     pgdb.reindex_concurrent()
                 else:
                     pgdb.reindex()
+
+
+@cmd
+def delete_room(path: "configuration file",
+                room: "room ID or alias",
+                debug: "enable debugging output"=False,
+                verbose: "enable verbose operation"=False,
+                pretend: "only show what would be done"=False,
+                keep_going: "keep going on purge timeouts"=False):
+    """Delete room."""
+    c, pgdb = _configure(path,
+                         open_database=True,
+                         require_database=True,
+                         debug=debug)
+    room_id = get_real_room_id(room)
+    if c.database:
```

Comment:
You are passing `require_database=True` above, and it will raise a `SystemExit` exception if a database connection is not configured. Therefore, there is no need to check for `c.database` here.